### PR TITLE
feat: add recurringEvent attribute to api/v2/event-types

### PR DESCRIPTION
**Pull request by ibex088**
### Overview

This PR introduces the `recurringEvent` attribute to the following updated endpoints:

#### Event Types
- **[POST]** `/v2/event-types`
- **[PATCH]** `/v2/event-types/{eventTypeId}`
- **[GET]** `/v2/event-types/{eventTypeId}` — now includes `recurringEvent`.

#### Organizations Event Types
- **[POST]** `/v2/organizations/{orgId}/teams/{teamId}/event-types`
- **[PATCH]** `/v2/organizations/{orgId}/teams/{teamId}/event-types/{eventTypeId}`
- **[GET]** `/v2/organizations/{orgId}/teams/{teamId}/event-types/{eventTypeId}` — now includes `recurringEvent`.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
- **Option 1:**
~~1. Set `"version": "1.2.3"` in `packages/platform/libraries/package.json`.~~
~~2. Add `"@calcom/platform-libraries-1.2.3": "npm:@calcom/platform-libraries@1.2.3"` to `apps/api/v2/package.json`.~~

- **Option 2:**
~~**⚠️ Important: Need to Publish the Platform libraries.**~~ 
✅ **Published 0.0.30**
  
---

### Testing `api/v2/event-types`:

1. **Recurring Event:**
Screenshot 2024-08-17 at 1 17 49 PM

   - **Input for the handler:**
     ```json
     {
       "recurringEvent": {
         "freq": 2,
         "interval": 8,
         "count": 10
       }
     }
     ```
   - **API Endpoint:**
     ```json
     {
       "recurrence": {
         "frequency": "weekly",
         "interval": 8,
         "occurrences": 10
       }
     }
     ```

---

### CODE REVIEW

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [inline packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:1]: We already had the `RecurringEvent_2024_06_14` class being used for output formatting (do a `RecurringEvent_2024_06_14` global search on `main`). I decided to reuse this class for input as well and removed some fields that I think were unnecessary.

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [inline packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:33]: removed

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [inline packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:29]: removed

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [inline packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:22]: renamed count to occurrences

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [reply packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:1]: also wanted to point out that - 
This change also modifies the output format, which could potentially cause breaking changes. I’ll wait for your feedback on this.

**Comment by Ryukemeister** [comment]
@SomayChauhan @supalarry just a heads up that atoms build is broken at the moment so ideally we should only merge this after #16248

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts
Comment (supalarry) [inline packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts:110]: What do you think about renaming `recurringEvent` to `recurrence` and `RecurringEvent_2024_06_14` to `Recurrence_2024_06_14` ?

I am thinking about it because the input class is already for an event and recurrance is a characteristic of an event, but recurringEvent to me sounds more like a higher level description like it could be used for class name for example instead of being used as a property name.

What do you think?

**Review by supalarry — COMMENTED** [review]
Great PR Somay! One comment:

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts
Comment (SomayChauhan) [reply packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts:110]: Sounds Good, will make changes

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts
Comment (SomayChauhan) [reply packages/platform/types/event-types/event-types_2024_06_14/inputs/recurring-event.input.ts:1]: Hi @Lucas-dOliveira,

For now, limiting the frequency options to yearly, monthly, and weekly since our front-end doesn't currently support daily frequency. 

Screenshot 2024-08-20 at 9 52 16 AM

If you'd like support for daily frequency, please raise an issue, and our team will review it.

File: packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts
Comment (SomayChauhan) [reply packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts]: @supalarry made the changes

File: apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.e2e-spec.ts
Comment (ThyMinimalDev) [inline apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.e2e-spec.ts:30]: please don't import from /dist folder, export the file inside of platform/types/index.js and import from "@calcom/platform-types" after rebuilding it

File: packages/lib/event-types/transformers/api-request.ts
Comment (ThyMinimalDev) [inline packages/lib/event-types/transformers/api-request.ts:19]: same here, import from "@calcom/platform-types" after exporting to index.js

File: packages/lib/event-types/transformers/api-response.ts
Comment (ThyMinimalDev) [inline packages/lib/event-types/transformers/api-response.ts:22]: sam here

File: packages/lib/event-types/transformers/api-response.ts
Comment (ThyMinimalDev) [inline packages/lib/event-types/transformers/api-response.ts:269]: not sure why we need unknown here ? is the TS not working? can't we directly do `satisfies Recurrence_2024_06_14`

File: packages/lib/event-types/transformers/api-request.ts
Comment (ThyMinimalDev) [inline packages/lib/event-types/transformers/api-request.ts:175]: satisfies TransformRecurringEventSchema_2024_06_14

File: packages/lib/event-types/transformers/api-request.ts
Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-request.ts:19]: Hi @ThyMinimalDev 
if we export the file inside of platform/types/index.js and import from "@calcom/platform-types" the build fails, 
not sure why.

File: packages/lib/event-types/transformers/api-response.ts
Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-response.ts]: made changes in https://github.com/calcom/cal.com/pull/16251/commits/126a7b7cbffc50825b3e4ada442db69221d4edd1

File: packages/lib/event-types/transformers/api-request.ts
Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-request.ts]: made changes in https://github.com/calcom/cal.com/pull/16251/commits/126a7b7cbffc50825b3e4ada442db69221d4edd1

File: packages/lib/event-types/transformers/api-request.ts
Comment (supalarry) [reply packages/lib/event-types/transformers/api-request.ts:19]: What if the enum is exported from `packages/platform/types/event-types/event-types_2024_06_14/inputs/enums/index.ts` and then imported as suggested?

Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-request.ts]: I made the suggested changes. in https://github.com/calcom/cal.com/pull/16251/commits/464d2611ed19dbd6ae390332b4d6dd5884bcae3b  However, I’m encountering an issue where the platform libraries build is failing.

File: packages/lib/event-types/transformers/api-request.ts
Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-request.ts]: Fixed

File: packages/lib/event-types/transformers/api-response.ts
Comment (SomayChauhan) [reply packages/lib/event-types/transformers/api-response.ts]: Fixed

**Comment by SomayChauhan** [comment]
✅ **Published 0.0.30**

**Review by supalarry — APPROVED** [review]
Great PR Somay! 💯

**Pull request merged**
